### PR TITLE
Adds PrintCsv() function and unit tests to bluemix/terminal/table

### PR DESCRIPTION
**Pull request by edsonarios**
I would like to add a new format to represent the tables in `ibmcloud sl` where we use them a lot, print in csv format

Using the same format
```go
testTable := NewTable(&buf, []string{"col1", "col2"})
testTable.Add("row1-col1", "row1-col2")
testTable.Add("row2-col1", "row2-col2")
testTable.PrintCsv()
```
The output would look like this
```
col1,col2
row1-col1,row1-col2
row2-col1,row2-col2
```

### CODE REVIEW

**Comment by steveclay** [comment]
Build is failing with: 
```
gosec -exclude-dir=plugin_examples -exclude-dir=vendor -quiet ./...
Results:
[/home/travis/gopath/src/github.com/IBM-Cloud/ibm-cloud-cli-sdk/bluemix/terminal/table.go:166] - G104 (CWE-703): Errors unhandled. (Confidence: HIGH, Severity: LOW)
    165: 	csvwriter.Write(t.headers)
  > 166: 	csvwriter.WriteAll(t.rows)
    167: }
[/home/travis/gopath/src/github.com/IBM-Cloud/ibm-cloud-cli-sdk/bluemix/terminal/table.go:165] - G104 (CWE-703): Errors unhandled. (Confidence: HIGH, Severity: LOW)
    164: 	csvwriter := csv.NewWriter(t.writer)
  > 165: 	csvwriter.Write(t.headers)
    166: 	csvwriter.WriteAll(t.rows)

**Comment by edsonarios** [comment]
Fixed build

**Comment by steveclay** [comment]
@edsonarios thank you for this PR. It is under consideration for inclusion to the SDK.

File: bluemix/terminal/table.go [hunk 1]
```diff
@@ -157,3 +159,15 @@ func (t *PrintableTable) PrintJson() {
 	// mimic behavior of Print()
 	t.rows = [][]string{}
 }
+
+func (t *PrintableTable) PrintCsv() {
```
Comment (tonystarkjr3) [inline bluemix/terminal/table.go:163]: I think we might want to have at least a couple of blocks of testing code on whether the error conditions are correct (e.g. some assertions on whether error output contains the text in lines 167/171 when the corresponding issues arise), to be better in-keeping with the other unit testing code. I can code this, or you can take a stab at it if you like @edsonarios.

Comment (tonystarkjr3) [reply bluemix/terminal/table.go:163]: It actually seems that the nature of our `table` interface is appropriately restrictive enough (strings-only parameters to functions, only three exposed functions, etc.) that there shouldn't ever really be a `Write`/`WriteAll` error (which occurs when the specified buffer--in this case, the `table`'s writer--cannot be flushed). So, the testing seems adequate here.

I'd just add one scenario to test whether an empty table is correspondingly printed as a zero-length string (because the API responses that our code outputs are occasionally empty). You can add it to the test file changed, and I think this'll be ready for merge!

```
import (
        ...
        "strings"
)

func TestEmptyTable(t *testing.T) {
	buf := bytes.Buffer{}
	testTable := NewTable(&buf, []string{})
	testTable.PrintCsv()
	assert.Equal(t, len(strings.TrimSpace(buf.String())), 0)
}
```

Comment (edsonarios) [reply bluemix/terminal/table.go:163]: The unit test was added for empty table, however seeing the code it's possible to test these errors, but it will be necessary added a new function for change the delimiter that by defect is ',' for a personalized one (example: ';' ':'), so we can cause the error, setting an invalid character like '\n', and so we can test it

However, I don't know if this feature will be necessary or very useful.

But for now the unit test was added and it's working in the new output format

Comment (tonystarkjr3) [reply bluemix/terminal/table.go:163]: I did think about changing the delimiter, but it seemed to me a case of going out of our way to think of an unrealistic scenario just to cause a parsing error. So, I agree with your comment that the usefulness of such a test is doubtful. I think we've covered our bases here.

File: bluemix/terminal/table.go [hunk 2]
```diff
@@ -157,3 +159,15 @@ func (t *PrintableTable) PrintJson() {
...
+	csvwriter := csv.NewWriter(t.writer)
+	err := csvwriter.Write(t.headers)
+	if err != nil {
+		fmt.Fprintln(t.writer, "Failed, header could not convert to csv format")
```
Comment (steveclay) [inline bluemix/terminal/table.go:167]: @edsonarios the only thing I will add here is that these error messages are not translated. Does csvwriter not return a suitable error message that can be output or even returned? Generally I would expect the operation to fail if you get errors on these calls to Write and WriteAll -- `err` just is returned and the logic does not continue.

Comment (edsonarios) [reply bluemix/terminal/table.go]: Was added translation for message error, also was returner the error when fail the functions Write and WriteAll with the personalized message

**Review by steveclay — APPROVED** [review]
